From 9066b0c318589f47b754a3def4fe8ec4688dc21a Mon Sep 17 00:00:00 2001
From: Chunming Zhou <david1.zhou@amd.com>
Date: Tue, 25 Aug 2015 15:12:26 +0800
Subject: [PATCH 0499/1050] drm/amdgpu: fix no sync_wait in copy_buffer

when eviction is happening, if don't handle
dependency, then the fence could be dead off.

Signed-off-by: Chunming Zhou <david1.zhou@amd.com>
Reviewed-by: Jammy Zhou <Jammy.Zhou@amd.com>
Reviewed-by: Christian K?nig <christian.koenig@amd.com>
---
 drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
index d7c02e1..4cb8132 100644
--- a/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
+++ b/drivers/gpu/drm/amd/amdgpu/amdgpu_ttm.c
@@ -1014,13 +1014,19 @@ int amdgpu_copy_buffer(struct amdgpu_ring *ring,
 	/* for fence and sync */
 	num_dw += 64 + AMDGPU_NUM_SYNCS * 8;
 
+	r = amdgpu_sync_wait(&sync);
+	if (r) {
+		DRM_ERROR("sync wait failed (%d).\n", r);
+		amdgpu_sync_free(adev, &sync, NULL);
+		return r;
+	}
+
 	r = amdgpu_ring_lock(ring, num_dw);
 	if (r) {
 		DRM_ERROR("ring lock failed (%d).\n", r);
 		amdgpu_sync_free(adev, &sync, NULL);
 		return r;
 	}
-
 	amdgpu_sync_rings(&sync, ring);
 
 	for (i = 0; i < num_loops; i++) {
-- 
1.9.1

